### PR TITLE
SAK-44557 Remove section and column merge link id from href

### PR DESCRIPTION
**Pull request by davidpbauer**
Section and column break item id is needed in javascript to remove the
section or column. This was added to the href of the link but should instead
be added to a data attribute on the link.

### CODE REVIEW

File: lessonbuilder/tool/src/webapp/templates/ShowPage.html [hunk 1]
```diff
@@ -512,7 +512,7 @@ <h3 rsf:id="sectionHeader" id="sectionHeader" class="sectionHeader"><span aria-h
 			  <div class="editsection">
 			  <span rsf:id="section-td" class="sectionedit">
                   <h3 class="offscreen" rsf:id="break-msg"></h3>
-                  <a role="button" rsf:id="section-del-link" href="#"  onclick="return false"><span aria-hidden="true" class="fa-compress fa-edit-icon"></span></a>
+                  <a role="button" rsf:id="section-del-link" href="javascript:void(0);"><span aria-hidden="true" class="fa-compress fa-edit-icon"></span></a>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply lessonbuilder/tool/src/webapp/templates/ShowPage.html:515]: @davidpbauer Could you add a tooltip and aria-label to this? I think this is a case of the a11y checker getting it right.

[the remark this answers is not in the document]
Comment (davidpbauer) [reply lessonbuilder/tool/src/webapp/templates/ShowPage.html:515]: Thanks! Title and aria-label are added in java in the showpageproducer.
`				    link.decorate(new UIFreeAttributeDecorator("title", messageLocator.getMessage("simplepage.join-items")));
				    link.decorate(new UIFreeAttributeDecorator("aria-label", messageLocator.getMessage("simplepage.join-items")));`

[the remark this answers is not in the document]
Comment (adrianfish) [reply lessonbuilder/tool/src/webapp/templates/ShowPage.html:515]: Right, I forgot about that, sorry.

**Review by adrianfish — APPROVED** [review]

File: lessonbuilder/tool/src/webapp/templates/ShowPage.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidpbauer) [reply lessonbuilder/tool/src/webapp/templates/ShowPage.html:515]: No worries. I didn't have the aria-label in there and just added it. So, thanks for saying something!

**Comment by davidpbauer** [comment]
After some research, it seems like the accepted conclusion is that an anchor tag with no location or resource should not be an anchor at all. Typically, this means that any time you find yourself needing to use href="javascript:;", href="#0", etc. it's better to use a button and style it however needed. I've modified the commit to also change the merge links to buttons as well as style them to match what they used to look like. This is also more accessible and better for keyboard users.

**Pull request merged**